typescript/types.ts

```diff
@@ -0,0 +1,36 @@
+export interface Board {
+    width: number,
+    height: number,
+    types: any[],
+    orbs: any[],
+    attic?: any,
+    generateOrbs(): void,
+    evaluate(): MatchData,
+    needsShuffle(): boolean,
+    hasPotentialMatch(): boolean,
+    hasMatch(): boolean,
+    unmatch(): void,
+    shuffle(): void
+}
+
+export type Orb = string | number
+
+export type MatchData = [any, number][]
+
+export interface BlanksBelow {
+	[coord: string]: number
+}
+
+export type Coord = [number, number]
+
+export interface PositionInfo {
+	position: {
+		first: Coord,
+		last: Coord
+	}
+}
+
+export type IterchunksWithPosition = {
+    [index: number]: Orb[];
+    positionInfo: PositionInfo[][]
+}
```

Comment:
I got this to work right (I think). I still need to update the types in `tools.iterchunks`, and `triples.find`, but it made me rethink the return type of `iterchunks`(yes it is still named that). I think it should be a list of objects that looks like this:
```
type WhatIterchunksReturns = {
    chunk: Orb[][],
    positionInfo: PositionInfo
}[]
```